src/icons/fileIcons.ts

```diff
@@ -1768,5 +1768,9 @@ export const fileIcons: FileIcons = {
       name: 'gridsome',
       fileNames: ['gridsome.config.js', 'gridsome.server.js'],
     },
+    {
+      name: 'octave',
+      fileExtensions: ['m']
```

Comment:
.m is not associated with objc MATLAB or octave
You need to add it as a language icon and let users control it